prow/cmd/deck/main_test.go

```diff
@@ -416,3 +419,145 @@ func TestHelp(t *testing.T) {
 	handleAndCheck()
 	handleAndCheck()
 }
+
+func TestListProwJobs(t *testing.T) {
+	templateJob := &prowapi.ProwJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "first",
+			Namespace: "prowjobs",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+		Spec: prowapi.ProwJobSpec{
+			Refs: &prowapi.Refs{
+				Org:  "org",
+				Repo: "repo",
+			},
+		},
+	}
+
+	var testCases = []struct {
```

Comment:
We are missing important scenarios:
* hiddenRepos of hidden-org still returns org/repo
* org/hidden-repo still returns org/repo
* a list of org/repo, org/hidden-repo, hidden-org/random-repo, another-org/another-repo only filters out the targeted repos